src/main.cpp

```diff
@@ -3898,7 +3898,13 @@ bool ContextualCheckBlock(
 
     // If this is initial block download and "verifypowonly" is set, we'll skip verifying the transactions
     if (IsInitialBlockDownload(chainparams) && GetBoolArg("-verifypowonly", false)) {
-        return true;
+        if (fCheckpointsEnabled) {
+            CBlockIndex *pindexLastCheckpoint = Checkpoints::GetLastCheckpoint(chainparams.Checkpoints());
+            if (pindexLastCheckpoint && pindexLastCheckpoint->GetAncestor(pindexPrev->nHeight) == pindexPrev) {
+                // This block is connecting to an ancestor of a checkpoint: disable script checks
```

Comment:
This was lifted from elsewhere when it was copy-pasted. The logic seems correct although the comment is now wrong. We're not just disabling script checks, we're accepting the block.

Reply:
Is the comparison right though? It seems to be comparing just the pointer value rather than using an operator== function, I don't know under what conditions that might fail. Ah okay it's just a linked list. LGTM.